wandb/sdk/artifacts/artifact.py

```diff
@@ -1781,6 +1781,9 @@ def _download_entry(
             _thread_local_api_settings.cookies = cookies
             _thread_local_api_settings.headers = headers
 
+            if require_nexus and entry.ref is None:
```

Comment:
@szymon-piechowicz-wandb I think moving this check here will make things faster (at least not slower than the current python benchmarks). Wdyt?

Reply:
Running a test on the benchmark vm using this dev branch to check as well. But just wanted to get your thoughts too :)